src/posthog-core.ts

```diff
@@ -1365,6 +1365,10 @@ export class PostHog {
         this.surveys.getActiveMatchingSurveys(callback, forceReload)
     }
 
+    checkSurveyEligibility(surveyId: string): { eligible: boolean; reason?: string } {
```

Comment:
I'd not increase the API surface since this is exactly what `canRenderSurvey` does already